ts/features/bonus/bpd/screens/optInPaymentMethods/OptInPaymentMethodsChoiceScreen.tsx

```diff
@@ -1,36 +1,160 @@
-import React from "react";
+import React, { useMemo, useState } from "react";
 import { SafeAreaView, ScrollView } from "react-native";
 import { View } from "native-base";
 import I18n from "../../../../../i18n";
 import BaseScreenComponent from "../../../../../components/screens/BaseScreenComponent";
 import { IOStyles } from "../../../../../components/core/variables/IOStyles";
 import { H1 } from "../../../../../components/core/typography/H1";
 import ButtonDefaultOpacity from "../../../../../components/ButtonDefaultOpacity";
+import FooterWithButtons from "../../../../../components/ui/FooterWithButtons";
+import {
+  confirmButtonProps,
+  disablePrimaryButtonProps,
+  errorBorderedButtonProps
+} from "../../../bonusVacanze/components/buttons/ButtonConfigurations";
+import { BlockButtonProps } from "../../../../../components/ui/BlockButtons";
+import {
+  RadioButtonList,
+  RadioItem
+} from "../../../../../components/core/selection/RadioButtonList";
+import { H4 } from "../../../../../components/core/typography/H4";
+import { Body } from "../../../../../components/core/typography/Body";
+import { IOPrimaryBadge } from "../../../../../components/core/IOBadge";
+import { useBottomSheetMethodsToDelete } from "../../components/optInStatus/BottomSheetMethodsToDelete";
+import { LabelSmall } from "../../../../../components/core/typography/LabelSmall";
+import {
+  navigateToOptInPaymentMethodsThankYouDeleteMethodsScreen,
+  navigateToOptInPaymentMethodsThankYouKeepMethodsScreen
+} from "../../navigation/actions";
+import { useNavigationContext } from "../../../../../utils/hooks/useOnFocus";
 
-const OptInPaymentMethodsChoiceScreen = () => (
-  // The void customRightIcon and customGoBack are needed to have a centered header title
-  <BaseScreenComponent
-    showInstabugChat={false}
-    goBack={false}
-    headerTitle={I18n.t("bonus.bpd.optInPaymentMethods.choice.header")}
-    customRightIcon={{
-      iconName: "",
-      onPress: () => true
-    }}
-    customGoBack={
-      <ButtonDefaultOpacity onPress={() => true} transparent={true} />
-    }
-  >
-    <SafeAreaView
-      style={IOStyles.flex}
-      testID={"OptInPaymentMethodsCashbackUpdate"}
-    >
-      <ScrollView style={[IOStyles.horizontalContentPadding]}>
-        <H1>{I18n.t("bonus.bpd.optInPaymentMethods.choice.title")}</H1>
-        <View spacer />
-      </ScrollView>
-    </SafeAreaView>
-  </BaseScreenComponent>
+type PaymentMethodsChoiceOptions =
+  | "keepPaymentMethods"
+  | "deletePaymentMethods";
+
+const generateOptionBody = (
+  title: string,
+  body: string,
+  showBadge?: boolean
+) => (
+  <>
+    {showBadge ? (
+      <>
+        <IOPrimaryBadge>
+          {I18n.t("bonus.bpd.optInPaymentMethods.choice.suggestedOption")}
+        </IOPrimaryBadge>
+        <View spacer xsmall />
+      </>
+    ) : undefined}
```

Comment:
```suggestion
    {showBadge && (
      <>
        <IOPrimaryBadge
          text={I18n.t("bonus.bpd.optInPaymentMethods.choice.suggestedOption")}
        />
        <View spacer xsmall />
      </>
    )}
```
I guess conditional render with `&&` is more readable when you have not an `else` case